src/coreclr/nativeaot/CMakeLists.txt

```diff
@@ -30,6 +30,8 @@ if(CLR_CMAKE_HOST_UNIX)
 
   add_compile_options(-fno-rtti)          # Native AOT runtime doesn't use RTTI
   add_compile_options(-fno-exceptions)    # Native AOT runtime doesn't use C++ exception handling
+  add_compile_options(-nostdlib)
```

Comment:
Would `-nostdlib++` be a safer option? What is kind of problem do you see when you do not pass this at all?

Reply:
`-nostdlib++` was not available on older versions of clang and gcc that's why I used `-nostdlib`.

> What is kind of problem do you see when you do not pass this at all?

This line is the whole point of the PR. 🙂 or did I misunderstood the question?

Reply:
I thought that they key part is removing `-lstdc++`. What are the additional things that `-nostdlib` does beyond that? 

The documentation suggests that `-nostdlib` is removing the dependency on all standard libraries, including C standard libraries and built-ins libraries. I am not sure what to expect from that.

Reply:
Right, we are building the archive without any C++ runtime dependency, therefore`-lstdc++` is not needed during the publishing of app; `-nostdlib` just makes it easier to figure out the dependencies on libstdc++.

In the scope of `libRuntime.WorkstationGC.a`, it leaves the external linkage for C APIs. When we compile it with `clang libRuntime.WorkstationGC.a <other-objects>` during the publishing of NativeAOT app, if we specify `-nostdlib` there, then the final binary will not have the libc linkage (but we will get tons of errors if we try to do that).

Currently the only difference before and after this PR is the absence of libstdc++ dependency in the NativeAOT published app.